module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -1971,23 +1978,37 @@ protected function getOAuth2Token($renew = false)
      * This is responsible for retrieving the status information of a certain
      * record.
      *
-     * @param string $id     The record id to retrieve the holdings for
-     * @param array  $patron Patron information, if available
+     * @param string $id      The record id to retrieve the holdings for
+     * @param array  $patron  Patron information, if available
+     * @param array  $options Extra options
      *
      * @return array An associative array with the following keys:
      * id, availability (boolean), status, location, reserve, callnumber.
      */
-    protected function getItemStatusesForBiblio($id, $patron = null)
+    protected function getItemStatusesForBiblio($id, $patron = null, array $options = [])
     {
-        $result = $this->makeRequest(
-            [
-                'path' => [
-                    'v1', 'contrib', 'kohasuomi', 'availability', 'biblios', $id,
-                    'search',
-                ],
-                'errors' => true,
-            ]
-        );
+        if (isset($options['itemLimit']) && $options['itemLimit'] > 0) {
```

Comment:
I would simplify this to:

```
        $requestParams = [
            'path' => [
                'v1', 'contrib', 'kohasuomi', 'availability', 'biblios', $id,
                'search',
            ],
            'errors' => true,
        ];
        if (($options['itemLimit'] ?? 0) > 0) {
            $requestParams['query'] => [ 'limit' => $options['itemLimit'] ];
        }
        $result = $this->makeRequest($requestParams);
```

Reply:
thanks @demiankatz I've added that in my follow-up!. I noticed the first => in             `$requestParams['query'] => [ 'limit' => $options['itemLimit'] ];` caused the unit tests to fail when I ran them locally with `vendor/bin/phing phpstan-console` so I changed it to a =

Then tested locally and the limit was being passed through to the Koha rest plugin successfully.

Please let me know if that change is incorrect?

Reply:
Yes, that was a typo, so your correction is correct. Sorry about that! The unit tests are doing their job! :-)

Reply:
Lovely, thank you!